File: python/acer/printer.py
import os

import pyfiglet
from tabulate import tabulate


class Printer:
    WELCOME = "Welcome to..."
    APP_NAME = "A.C.E.R."
    NUMBER_OF_PLAYERS_PROMPT = "Number of players? "
    NUMBER_OF_DAYS_PROMPT = "Number of days in calendar? "
    RESULTS = "Results..."
    READY_PROMPT = "Ready? Here we go... "
    FAREWELL = "Thanks for using A.C.E.R.! Enjoy your advent calendar!"

    def print_opening_screen(self):
        self.clear_screen()
        self.print_welcome()
        self.print_logo()

    def print_final_results(self, game):
        self.clear_screen()
        self.print_logo()
        self.print_results()
        self.print_player_table(game)
        self.print_farewell()

    @staticmethod
    def clear_screen():
        os.system('cls' if os.name == 'nt' else 'clear')

    def print_welcome(self):
        print(self.WELCOME)

    def print_logo(self):
        print()
        print(pyfiglet.figlet_format(self.APP_NAME, font="banner3-D"))
        print(" ~ Advent        Calendar       Enjoyment      Randomizer ~")
        print()

    def print_ready(self):
        print(self.READY_PROMPT)
        print()

    def print_results(self):
        print(self.RESULTS)

    @staticmethod
    def print_player_table(game):
        headers = ["Player", "Days to Open"]
        table = []

        for p in game.players:
            row = [p.name, p.final_results()]
            table.append(row)

        print(tabulate(table, headers, tablefmt="fancy_grid"))
        print()

    def print_farewell(self):
        print(self.FAREWELL)

    def print_number_of_players_prompt(self):
        print(self.NUMBER_OF_PLAYERS_PROMPT)

    def print_number_ofe_days_prompt(self):
        print(self.NUMBER_OF_DAYS_PROMPT)

    @staticmethod
    def print_player_name_prompt(i):
        print(f"Name of Player {i}? ")

    @staticmethod
    def print_player_turn_result(player_name, day_to_open):
        print(f"{player_name}, you got day {day_to_open}")
        print()

File: golang/printer.go
package main

import (
	"fmt"
	"os"

	"github.com/common-nighthawk/go-figure"
	"github.com/inancgumus/screen"
	"github.com/olekukonko/tablewriter"
)

const (
	welcome               = "Welcome to..."
	appName               = "A.C.E.R."
	numberOfPlayersPrompt = "Number of players? "
	numberOfDaysPrompt    = "Number of days in calendar? "
	results               = "Results..."
	readyPrompt           = "Ready? Here we go... "
	farewell              = "Thanks for using A.C.E.R.! Enjoy your advent calendar!"
)

type Printer struct {
}

func (p Printer) PrintOpeningScreen() {
	clearScreen()
	printWelcome()
	printLogo()
	printReady()
}

func (p Printer) PrintFinalResults(game Game) {
	clearScreen()
	printLogo()
	printResults()
	p.PrintPlayerTable(game)
	printFarewell()
}

func clearScreen() {
	screen.Clear()
}

func printWelcome() {
	fmt.Println(welcome)
}

func printLogo() {
	fmt.Println()
	myFigure := figure.NewFigure(appName, "isometric1", true)
	myFigure.Print()
	fmt.Println("( Advent           Calendar          Enjoyment        Randomizer )")
	fmt.Println()
}

func printReady() {
	fmt.Println(readyPrompt)
}

func (p Printer) PrintNumberOfPlayersPrompt() {
	fmt.Print(numberOfPlayersPrompt)
}

func (p Printer) PrintNumberOfDaysPrompt() {
	fmt.Print(numberOfDaysPrompt)
}

func (p Printer) printRequestPlayerInteraction(currentPlayerName string) {
	fmt.Printf("%s, hit <enter> to pick a number... ", currentPlayerName)
}

func (p Printer) printPlayerTurnResult(playerName string, dayToOpen int) {
	fmt.Printf("%s, you got day %d", playerName, dayToOpen)
	fmt.Println()
}

func (p Printer) printDaysRemaining(daysRemaining int) {
	if daysRemaining == 0 {
		fmt.Println("\n[All Days Picked!]")
	} else {
		fmt.Printf("[Days remaining: %d]\n", daysRemaining)
	}
	fmt.Println()
}

func printResults() {
	fmt.Println(results)
}

func (p *Printer) PrintPlayerTable(game Game) {
	table := tablewriter.NewWriter(os.Stdout)
	table.SetHeader([]string{"Player", "Days to Open"})

	for _, p := range game.Players {
		var row = []string{p.name, p.finalResults()}
		table.Append(row)
	}

	table.Render()
	fmt.Println()
}

func printFarewell() {
	fmt.Println()
	fmt.Println(farewell)
}
